packages/state-manager/src/sagas/users/userProfile/saveUserProfile.saga.ts

```diff
@@ -0,0 +1,68 @@
+import { type PayloadAction } from '@reduxjs/toolkit'
+import { call, select, apply, put } from 'typed-redux-saga'
+import { arrayBufferToString } from 'pvutils'
+import * as Block from 'multiformats/block'
+import * as dagCbor from '@ipld/dag-cbor'
+import { sha256 } from 'multiformats/hashes/sha2'
+
+import { signData, loadPrivateKey, pubKeyFromCsr } from '@quiet/identity'
+import { UserProfile, UserProfileData, SocketActionTypes } from '@quiet/types'
+
+import { config } from '../../users/const/certFieldTypes'
+import { identitySelectors } from '../../identity/identity.selectors'
+import logger from '../../../utils/logger'
+import { usersActions } from '../users.slice'
+import { type Socket, applyEmitParams } from '../../../types'
+
+const log = logger('saveUserProfileSaga')
+
+const fileToBase64String = (file: File): Promise<string> => {
+  return new Promise<string>((resolve, reject) => {
+    const reader = new FileReader()
+
+    reader.onload = () => {
+      if (reader.result) {
+        resolve(reader.result as string)
+      } else {
+        reject(new Error(`Failed to read file, result: ${reader.result}`))
+      }
+    }
+    reader.onerror = reject
+    reader.readAsDataURL(file)
+  })
+}
+
+export function* saveUserProfileSaga(socket: Socket, action: PayloadAction<{ photo?: File }>): Generator {
```

Comment:
I think I'd be good to add test for this saga

Reply:
I've tested this manually. I suggest that we consider adding a test in a separate PR. I would like to refactor this by moving it to the backend soon.

Reply:
Generally we want to make sure the tests prove the feature is correct, before we deploy. 

If we think there's a refactor imminent, maybe we could write a test that would work across the refactor, such as an e2e test? Since we don't know when we'll do the refactor, I don't think it's a good reason to stop using testing as we have been.

Reply:
Talking with @holmesworcester, it sounds like the team has made a decision to go all in on automated testing, so I'll add a test here

Reply:
It's not only the team policy but also one important thing about sagas (and one of the main reasons we use them) is that they're extremely easy to test so it comes with a very low cost